api/v1beta2/moduletemplate_types.go

```diff
@@ -78,6 +78,18 @@ type ModuleTemplateSpec struct {
 	// +kubebuilder:validation:MinLength:=3
 	Channel string `json:"channel"`
 
+	// Version identifies the version of the Module. It can be empty, or a semantic version.
+	// +optional
+	// +kubebuilder:validation:Pattern:=^((0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)?)?$
+	// +kubebuilder:validation:MaxLength:=32
+	Version string `json:"version"`
+
+	// ModuleName is the name of the Module. Can be empty.
+	// +optional
+	// +kubebuilder:validation:Pattern:=^([a-z][a-z][a-z]+(-[a-z][a-z][a-z]+)*)?$
```

Comment:
Maybe this alternative would be a bit simpler to read `^([a-z]{3,}(-[a-z]{3,})*)?$`